source/docs/advanced/theme/route-dispatcher.md

```diff
@@ -22,103 +22,118 @@ If you come from a Magento background, this is the concept behind [URL Rewrites]
 </blockquote>
 
-The goal here will be [to override the resolver](/docs/reference/graphql-module-definition.html#resolvers-optional) of `route` in order to add your own URLs.
+To add a new `urlMatcher` to a module you need to register `FrontCommerce/Core` as a module dependecy and then use `registerUrlMatcher` to add a custom `urlMatcher`:
+
+```diff
+import typeDefs from "./schema.gql";
++import myCustomUrlMatcher from './urlMatcher.js';
+import MyCustomLoader from './loader.js';
+import { makeUserClientFromRequest } from "server/modules/magento2/core/factories";
+
+const moduleDefinition = {
+  namespace: "MyCustomModule",
+-  dependencies: ["Front-Commerce/Core"],
++  dependencies: [],
+  typeDefs,
+  contextEnhancer: ({ req, loaders }) => {
+    const axiosInstance = makeUserClientFromRequest(req);
+    const CustomLoader = MyCustomLoader(axiosInstance);
++    loaders.Page.registerUrlMatcher(myCustomUrlMatcher(CustomLoader));
+    return {
+      MyCustomLoader: CustomLoader
+    };
+  },
+};
 
-1. Create a custom GraphQL module that will add new `resolvers` to your GraphQL Schema See [Create a new GraphQL module](/docs/essentials/extend-the-graphql-schema.html) for more details.
-2. Create a loader that will match a URL string with an Entity (the object returned below):
-```js
-const MyModuleUrlLoader = makeDataLoader => () => {
-  return {
-    load: url => {
-      if (url === "my-dynamic-custom-url") {
-        return {
-          // The url that you are currently matching
-          url: "my-dynamic-custom-url",
-          // The type of entity you are willing to
-          // display for this URL
-          type: "myPageType",
-          // The id that will let you load the
-          // correct entity within your page component
-          identifier: "identifier",
-          // The path that will let you know what's
-          // the supposed URL before the URL rewrite
-          target_path: "catalog/product/view/id/17",
-          // A type that will let you know if its the
-          // correct URL or if you should redirect
-          // it. If it is a redirection, it should have
-          // the value `301`.
-          redirect_type: 0
-        }
-      }
-    }
-  }
-}
+export default moduleDefinition;
 ```
-    <blockquote class="note">
-    See [Loaders](/docs/advanced/graphql/slim-down-resolvers-with-loaders.html) documentation to learn how to instantiate your loader and add it to your GraphQL context.
-    </blockquote>
 
-    Here, we only handle `my-dynamic-custom-url` URL. However, you will most likely need to fetch the result from one of your backend instead. If you do so, the final result should still match the structure of an Entity as presented above.
+```diff
+// urlMatcher.js
++const myCustomUrlMatcher = (loader) => async (url, match) => {
++  if(match.__typename) {
```

Comment:
CS in this example needs to be reworked.
We must try to keep it as standard as possible.

I suggest at least:
- pass it into Prettier (either in your editor or in https://prettier.io/playground/)
- ensure the code is _valid_ (e.g: `<canonical url>` should instead use a property of `myMatch`)
- prevent looooooong lines: either remove comments (over a paragraph below) or put them on another line(s)… refactoring into temporary variables can also make things more expressive and remove the need for some comments